app/src/main/java/edu/mills/heartoftaiwanese/activity/home/HomeFragment.kt

```diff
@@ -0,0 +1,213 @@
+package edu.mills.heartoftaiwanese.activity.home
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import com.google.android.material.tabs.TabLayout
+import edu.mills.heartoftaiwanese.R
+import edu.mills.heartoftaiwanese.activity.BaseFragment
+import edu.mills.heartoftaiwanese.activity.hideKeyboard
+import edu.mills.heartoftaiwanese.databinding.FragmentHomeBinding
+import edu.mills.heartoftaiwanese.network.WebResultCode
+import java.util.Calendar
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [HomeFragment.newInstance] factory method to
```

Comment:
You call the constructor from other classes instead of following this advice. Either remove the comment or make the constructor private.